level1c4pps/isccpng2pps_lib.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 level1c4pps developers
+#
+# This file is part of level1c4pps
+#
+# level1c4pps is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# level1c4pps is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with level1c4pps.  If not, see <http://www.gnu.org/licenses/>.
+# Author(s):
+
+#   Nina Hakansson <nina.hakansson@smhi.se>
+
+"""Functions to convert ISCCP Next Generation level-1-G data to a NWCSAF/PPS level-1c formatet netCDF/CF file."""
+
+import os
+import time
+import xarray as xr
+import numpy as np
+from satpy.scene import Scene
+from level1c4pps import (get_encoding,
+                         dt64_to_datetime,
+                         compose_filename,
+                         apply_sunz_correction,
+                         rename_latitude_longitude,
+                         update_angle_attributes, get_header_attrs,
+                         set_header_and_band_attrs_defaults,
+                         convert_angles,
+                         adjust_lons_to_valid_range)
+
+import logging
+
+
+logger = logging.getLogger('isccpng2pps')
+
+BANDNAMES = ['refl_01_60um',
+             'refl_00_65um',
+             'refl_00_86um',
+             'temp_03_80um',
+             'temp_08_60um',
+             'temp_09_70um',
+             'temp_11_00um',
+             'temp_12_00um',
+             'temp_13_30um',
+             'temp_06_20um',
+             'temp_07_30um']
+
+
+REFL_BANDS = ['refl_01_60um',
+              'refl_00_65um',
+              'refl_00_86um']
+
+ANGLE_NAMES = ['solar_zenith_angle', 'satellite_zenith_angle',
+               'solar_azimuth_angle', 'satellite_azimuth_angle']
+
+PPS_TAGNAMES = {'refl_01_60um': "ch_r16",
+                'refl_00_65um': "ch_r06",
+                'refl_00_86um': "ch_r09",
+                'temp_03_80um': "ch_tb37",
+                'temp_08_60um': "ch_tb85",
+                'temp_09_70um': "ch_tbxx",
+                'temp_11_00um': "ch_tb11",
+                'temp_12_00um': "ch_tb12",
+                'temp_13_30um': "ch_tb133",
+                'temp_06_20um': "ch_tbxx",
+                'temp_07_30um': "ch_tb73"}
+
+BANDNAMES = list(PPS_TAGNAMES.keys())
+
+coef_slope_chan = ['refl_00_65um', 'refl_00_86um', 'refl_01_60um',
```

Comment:
Maybe nicer to have put all coefficients related stuff in a yaml file with a format something like this:
```
coeffcients:
  channels: 
    - refl_00_65um 
    - refl_00_86um
    - refl_01_60um  
    - temp_03_80um
    - temp_06_20um
    - temp_07_30um
    - temp_08_60um  
    - temp_09_70um
    - temp_11_00um
    - temp_12_00um
    - temp_13_30um'
  satellite: 
    GOES-16: 
      day:
        slope: [1.00152, 0.946758, 1.00669, 0.926764, 0.991329, 0.999300, 1.01977, 1.03453, 0.996781, 1.02343, 0.863748]
        offset: [0.000369844, 0.193868, 0.708022, 18.4795, 1.58729, -0.0312020, -4.93224, -7.83331, 1.13395, -5.51940, 30.5574]
      night: 
        ...
```